### PR TITLE
Add formatting checks for Scala and Java

### PR DESCRIPTION
**Pull request by hubertp**
[ci no changelog needed]

### Pull Request Description

Contribution notes mention that formatting should be run prior to PR merges
but that is clearly not the case (see `Important Notes` section). 
Added a new workflow that runs after regular builds finish (and
succeed). Mostly c&p from scala workflow + `javafmtCheckAll` and
`scalafmtCheckAll` sbt commands. Translation to new CI workflow can be done independently.

### Important Notes
To illustrate the problem:
```
sbt:enso> javafmtCheck
...
[error] 66 files must be formatted
[error] 1 file must be formatted
[error] (runtime / Compile / javafmtCheck) 66 files must be formatted
[error] (logging-utils / Compile / javafmtCheck) 1 file must be formatted
[error] Total time: 2 s, completed Apr 12, 2022, 3:31:30 PM
```
and 
```
sbt:enso> scalafmtCheckAll
[warn] scalafmt: /home/hubert/work/repos/enso/enso/lib/scala/text-buffer: 1 files aren't formatted properly:
...
[warn] scalafmt: /home/hubert/work/repos/enso/enso/lib/scala/pkg: 1 files aren't formatted properly:
...
[warn] scalafmt: /home/hubert/work/repos/enso/enso/lib/scala/syntax/definition/.jvm: 1 files aren't formatted properly:
...
[warn] scalafmt: /home/hubert/work/repos/enso/enso/engine/polyglot-api: 1 files aren't formatted properly:
...
[warn] scalafmt: /home/hubert/work/repos/enso/enso/engine/runtime: 6 files aren't formatted properly:
...
[warn] scalafmt: /home/hubert/work/repos/enso/enso/engine/language-server: 2 files aren't formatted properly:
...
[warn] scalafmt: /home/hubert/work/repos/enso/enso/lib/scala/project-manager: 1 files aren't formatted properly:
...
[warn] scalafmt: /home/hubert/work/repos/enso/enso/engine/language-server: 8 files aren't formatted properly:
...
[info] scalafmt: Checking 127 Scala sources (/home/hubert/work/repos/enso/enso/engine/runtime)...
[warn] scalafmt: /home/hubert/work/repos/enso/enso/engine/runtime/src/main/scala/org/enso/compiler/codegen/AstView.scala isn't formatted properly!
[warn] scalafmt: /home/hubert/work/repos/enso/enso/engine/runtime/src/main/scala/org/enso/compiler/codegen/IrToTruffle.scala isn't formatted properly!
[error] stack trace is suppressed; run last runtime / Compile / scalafmtCheck for the full output
[error] scalafmt: 6 files must be formatted (/home/hubert/work/repos/enso/enso/engine/runtime)
[error] scalafmt: 1 files must be formatted (/home/hubert/work/repos/enso/enso/lib/scala/project-manager)
[error] scalafmt: 1 files must be formatted (/home/hubert/work/repos/enso/enso/lib/scala/syntax/definition/.jvm)
[error] scalafmt: 8 files must be formatted (/home/hubert/work/repos/enso/enso/engine/language-server)
[error] scalafmt: 2 files must be formatted (/home/hubert/work/repos/enso/enso/engine/language-server)
[error] scalafmt: 1 files must be formatted (/home/hubert/work/repos/enso/enso/lib/scala/text-buffer)
[error] scalafmt: 1 files must be formatted (/home/hubert/work/repos/enso/enso/engine/polyglot-api)
[error] scalafmt: 1 files must be formatted (/home/hubert/work/repos/enso/enso/lib/scala/pkg)
[error] (runtime / Compile / scalafmtCheck) org.scalafmt.sbt.ScalafmtSbtReporter$ScalafmtSbtError: scalafmt: /home/hubert/work/repos/enso/enso/engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala:2052: error: ; expected but string constant found
[error]       override def showCode(indent: Int): String = s"\"$text\""
[error]                                                              ^ [/home/hubert/work/repos/enso/enso/engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala]
[error] (runtime / Test / scalafmtCheck) scalafmt: 6 files must be formatted (/home/hubert/work/repos/enso/enso/engine/runtime)
[error] (project-manager / Compile / scalafmtCheck) scalafmt: 1 files must be formatted (/home/hubert/work/repos/enso/enso/lib/scala/project-manager)
[error] (syntax-definition / Compile / scalafmtCheck) scalafmt: 1 files must be formatted (/home/hubert/work/repos/enso/enso/lib/scala/syntax/definition/.jvm)
[error] (language-server / Compile / scalafmtCheck) scalafmt: 8 files must be formatted (/home/hubert/work/repos/enso/enso/engine/language-server)
[error] (language-server / Test / scalafmtCheck) scalafmt: 2 files must be formatted (/home/hubert/work/repos/enso/enso/engine/language-server)
[error] (text-buffer / Compile / scalafmtCheck) scalafmt: 1 files must be formatted (/home/hubert/work/repos/enso/enso/lib/scala/text-buffer)
[error] (polyglot-api / Test / scalafmtCheck) scalafmt: 1 files must be formatted (/home/hubert/work/repos/enso/enso/engine/polyglot-api)
[error] (pkg / Compile / scalafmtCheck) scalafmt: 1 files must be formatted (/home/hubert/work/repos/enso/enso/lib/scala/pkg)
```

This is not the problem with individual contributions. Those things will slip through no matter how much one tries to do due diligence with the checklist below. Only automatic check will ensure the consistency.

### Checklist

Please include the following checklist in your PR:

- [ ] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed: Enso GUI was tested when built using BOTH `./run dist` and `./run watch`.

### CODE REVIEW

File: .github/workflows/formatting.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+          curl --retry 4 --retry-connrefused -fsSL -o lib/scala/pkg/src/main/resources/orders/data/store_data.xlsx https://github.com/enso-org/project-templates/raw/main/Orders/data/store_data.xlsx
+          curl --retry 4 --retry-connrefused -fsSL -o lib/scala/pkg/src/main/resources/orders/src/Main.enso https://github.com/enso-org/project-templates/raw/main/Orders/src/Main.enso
+          curl --retry 4 --retry-connrefused -fsSL -o lib/scala/pkg/src/main/resources/restaurants/data/la_districts.csv https://github.com/enso-org/project-templates/raw/main/Restaurants/data/la_districts.csv
+          curl --retry 4 --retry-connrefused -fsSL -o lib/scala/pkg/src/main/resources/restaurants/data/restaurants.csv https://github.com/enso-org/project-templates/raw/main/Restaurants/data/restaurants.csv
+          curl --retry 4 --retry-connrefused -fsSL -o lib/scala/pkg/src/main/resources/restaurants/src/Main.enso https://github.com/enso-org/project-templates/raw/main/Restaurants/src/Main.enso
+          curl --retry 4 --retry-connrefused -fsSL -o lib/scala/pkg/src/main/resources/stargazers/src/Main.enso https://github.com/enso-org/project-templates/raw/main/Stargazers/src/Main.enso
```
Comment (4e6) [inline .github/workflows/formatting.yml]: This step is required for packaging (and maybe running the tests). We can skip it here

File: .github/workflows/formatting.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
+name: Code Formatting Check
+
+on:
+  workflow_run:
+    workflows: [ "Engine CI (legacy)", "Engine CI (new)" ]
+    branches: [ main ]
+    types:
+      - completed
```
Comment (mwu-tow) [inline .github/workflows/formatting.yml]: I'd run this regardless of the primary build.

File: .github/workflows/formatting.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  push:
+    branches: [develop, "release/*"]
+  pull_request:
+    branches: ["*"]
```
Comment (mwu-tow) [inline .github/workflows/formatting.yml:7]: Please check if removing the branches key altogether works. Likely `*` does not match branch names with `/`.

File: .github/workflows/formatting.yml [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+      - name: Install FlatBuffers Compiler
+        run: conda install --freeze-installed flatbuffers=1.12.0
+      - name: Setup GraalVM Environment
+        uses: ayltai/setup-graalvm@v1
+        with:
+          graalvm-version: ${{ env.graalVersion }}
+          java-version: ${{ env.javaVersion }}
+          native-image: true
+      - name: Install Node
+        uses: actions/setup-node@v1
```
Comment (mwu-tow) [inline .github/workflows/formatting.yml]: Node is AFAIK used only by GUI.

File: .github/workflows/formatting.yml [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      - name: Install Dependencies of the Simple Library Server
+        shell: bash
+        working-directory: tools/simple-library-server
+        run: |
+          npm install
+      - name: Install Dependencies of the Simple Library Server
+        shell: bash
+        working-directory: tools/simple-library-server
+        run: |
+          npm install
```
Comment (mwu-tow) [inline .github/workflows/formatting.yml]: This is part of test setup.

File: .github/workflows/formatting.yml [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+      - name: Bootstrap Enso project
+        run: |
+          sleep 1
+          sbt --no-colors bootstrap
+
+      # Compile
+      - name: Build Enso
+        run: |
+          sleep 1
+          sbt --no-colors compile
```
Comment (mwu-tow) [inline .github/workflows/formatting.yml]: Please verify if these are actually needed.

File: .github/workflows/formatting.yml [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+          path: |
+            ~/.sbt
+            ~/.ivy2/cache
+            ~/.cache
+          key: Linux-sbt-${{ hashFiles('**build.sbt') }}
+          restore-keys: Linux-sbt-
+
+      - name: Bootstrap Enso project
+        run: |
+          sleep 1
```
Comment (mwu-tow) [inline .github/workflows/formatting.yml]: Please remove sleeps.

File: .github/workflows/formatting.yml [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+      # The actual check
+      - name: Check Scala formatting
+        run: |
+          sleep 1
+          sbt --no-colors scalafmtCheckAll
+      - name: Check Java formatting
+        run: |
+          sleep 1
+          sbt --no-colors javafmtCheckAll
```
Comment (mwu-tow) [inline .github/workflows/formatting.yml]: If possible, please try batching sbt commands into one.

**Review by mwu-tow — CHANGES_REQUESTED** [review]

File: .github/workflows/formatting.yml [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  pull_request:
+    branches: ["*"]
+
+env:
+  # Please ensure that this is in sync with graalVersion in build.sbt
+  graalVersion: 21.3.0
+  # Please ensure that this is in sync with javaVersion in build.sbt
+  javaVersion: 11
+  # Please ensure that this is in sync with project/build.properties
+  sbtVersion: 1.5.2
```
Comment (4e6) [inline .github/workflows/formatting.yml:15]: we have `1.6.1` in biuld properties. I think it's worth updating it for all workflows

**Review by 4e6 — APPROVED** [review]
Looks good. Maybe SBT version update is worth it's own PR (in case it breaks something)

File: .github/workflows/formatting.yml (hunk 9 above)
Comment (hubertp) [reply .github/workflows/formatting.yml:15]: Yes, I basically c&p existing workflows.

**Review by jdunkerley — APPROVED** [review]

**Review by mwu-tow — APPROVED** [review]

**Pull request merged**